dataflow/src/main/java/org/checkerframework/dataflow/cfg/CFGVisualizer.java

```diff
@@ -48,116 +49,135 @@
+    String visualizeStoreFieldVals(FlowExpressions.FieldAccess fieldAccess, A value);
 
     /**
      * Called by {@code CFAbstractStore#internalVisualize()} to visualize the value of arrays
-     * collected by this Store.
+     * collected by this store.
      *
      * @param arrayValue the array
      * @param value the value of the array
+     * @return the String representation of the arrays collected by this store
      */
-    void visualizeStoreArrayVal(FlowExpressions.ArrayAccess arrayValue, A value);
+    String visualizeStoreArrayVal(FlowExpressions.ArrayAccess arrayValue, A value);
 
     /**
      * Called by {@code CFAbstractStore#internalVisualize()} to visualize the value of pure method
-     * calls collected by this Store.
+     * calls collected by this store.
      *
      * @param methodCall the pure method call
      * @param value the value of the pure method call
+     * @return the String representation of the pure method calls collected by this store
      */
-    void visualizeStoreMethodVals(FlowExpressions.MethodCall methodCall, A value);
+    String visualizeStoreMethodVals(FlowExpressions.MethodCall methodCall, A value);
 
     /**
      * Called by {@code CFAbstractStore#internalVisualize()} to visualize the value of class names
-     * collected by this Store.
+     * collected by this store.
      *
      * @param className the class name
      * @param value the value of the class name
+     * @return the String representation of the class names collected by this store
      */
-    void visualizeStoreClassVals(FlowExpressions.ClassName className, A value);
+    String visualizeStoreClassVals(FlowExpressions.ClassName className, A value);
 
     /**
      * Called by {@code CFAbstractStore#internalVisualize()} to visualize the specific information
-     * collected according to the specific kind of Store. Currently, these Stores call this method:
+     * collected according to the specific kind of store. Currently, these Stores call this method:
      * {@code LockStore}, {@code NullnessStore}, and {@code InitializationStore} to visualize
      * additional information.
      *
      * @param keyName the name of the specific information to be visualized
      * @param value the value of the specific information to be visualized
+     * @return the String representation of the specific information according to the specific kind
+     *     of store
      */
-    void visualizeStoreKeyVal(String keyName, Object value);
+    String visualizeStoreKeyVal(String keyName, Object value);
 
     /**
      * Called by {@code CFAbstractStore} to visualize any information after the invocation of {@code
      * CFAbstractStore#internalVisualize()}.
+     *
+     * @return the String representation of the footer to use for a store
      */
-    void visualizeStoreFooter();
+    String visualizeStoreFooter();
 
     /**
-     * Visualize a block based on the analysis.
+     * Visualize a special block based on the analysis.
```

Comment:
Why did you add "special" here?  I expected it in `visualizeSpecialBlock`, but I am surprised to see it here in `visualizeBlock`.

Reply:
It's my fault, I will fix it.